integration_test/third_party_apps_test.go

```diff
@@ -172,65 +172,61 @@ func installAgent(ctx context.Context, logger *logging.DirectoryLogger, vm *gce.
 	return nonRetryable, agents.InstallPackageFromGCS(ctx, logger, vm, agents.OpsAgentType, packagesInGCS)
 }
 
-// expectedLogs encodes a series of assertions about what data we expect
-// to see in the logging backend.
-type expectedLogs struct {
-	// Note on tags: the "yaml" tag specifies the name of this field in the
-	// .yaml file.
-	LogEntries []expectedLog `yaml:"log_entries"`
+type logFields struct {
+	Name        string `yaml:"name" validate:"required"`
+	Value       string `yaml:"value" validate:"required"`
+	Type        string `yaml:"type" validate:"required"`
+	Description string `yaml:"description" validate:"optional,omitempty"`
 }
+
 type expectedLog struct {
-	LogName string `yaml:"log_name"`
-	// Map of field name to a regex that is expected to match the field value.
-	// For example, {"jsonPayload.message": ".*access denied.*"}.
-	FieldMatchers map[string]string `yaml:"field_matchers"`
+	LogName string       `yaml:"log_name" validate:"required"`
+	Fields  []*logFields `yaml:"fields" validate:"required"`
```

Comment:
These don't need to be pointers

Reply:
hmmm initially i thought it might be more efficient to use pointer. codes in confgenerator seem to all use pointer. As the struct is not actually big one, maybe i am being over optimized.